Selenium refactoring

jeddcuison opened the pull request:
Changes for this pull request is generally  to refactor the existing selenium tests. The objective is to establish a better way of coding xpath expressions, optimizing test execution, introduction of test suite to group tests that have dependencies with one another and to make selenium UI testing more robust and reliable.

While working on this, some of the tests exposed problems in the UI that is why one would notice 2 UI commits which is about

- stomp 
- loading icon

Those commits where cherry picked from the old repository which is the reason why the commit dates are old. In addition one would notice that the selenium commit proper is just one commit. It's also because those changes were from an the old repository but instead of cherry picking each and everyone of the commits, I just opted to copy the codes since they all reside in the selenium module.

jeddcuison committed:
Fix bug wherein if user clicks Configuration when stomp client is still connecting, configuration sub tabs do not show up

jeddcuison committed:
Optimize busy icon

jeddcuison committed:
Refactor selenium test module

- create test suites to group tests that are dependent on one another
- optimize selenium test executions by removing unnecessary waiting e.g. Thread.sleep
- establish baseline for more maintainable xpath expressions
- general code cleanup

akinny commented on jwala-integration-test/build.gradle:
```diff
@@ -12,7 +12,6 @@ dependencies {
-            "selenium.property.file": System.getProperty("selenium.property.file")
-    ]
+    systemProperty "selenium.property.path", System.getProperty("selenium.property.path")
+    exclude '**/testSuiteClasses/**'
```
Shouldn't this be  exclude '**/testSuite/**'?

jeddcuison replied:
Done renamed to testsuite

jmahmoodcc commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/testSuite/JwalaTest.java:
```diff
@@ -0,0 +1,62 @@
+ * Wraps Jwala specific UI actions e.g. click tab, etc also contains methods used commonly in tests
+ * Created by Jedd Cuison on 2/28/2017
+ */
+public class JwalaTest {
```
Since we're hard coding the Chrome Web Driver in this class we should modify the name of the class to indicate that it currently only works with Chrome

jeddcuison replied:
Good catch! I'll make it generic. I don't want to make JwalaTest specific to Chrome since it is inherited by all test suite classes.

jeddcuison replied:
DONE

jmahmoodcc commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/testSuite/JwalaTest.java:
```diff
@@ -0,0 +1,62 @@
+ * Wraps Jwala specific UI actions e.g. click tab, etc also contains methods used commonly in tests
+ * Created by Jedd Cuison on 2/28/2017
+ */
+public class JwalaTest {
```
Also this appears to be a utility class, so I would remove Test from the name as well

jmahmoodcc replied:
Oh nevermind, I see that your extending this class for the other tests now

jmahmoodcc reviewed (changes requested):
Before merging I'm requesting that the external properties tests and the history pop-up tests be recreated with the new testing structure. These were originally created because of failing regression tests that no one noticed for a long time, so we wanted to automate those tests. I like all the changes we made, but we still need to honor the gaps we had already closed with the initial implementation.

jeddcuison committed:
Remove unused import

jeddcuison committed:
Make JwalaTest generic

jeddcuison committed:
Rename class

jeddcuison committed:
Rename package testSuiteClass to testsuite

jeddcuison committed:
Test history table pop up

jeddcuison committed:
Remove "throws InterruptedException"

jeddcuison committed:
Fix bug wherein external property upload fails and add tests to verify the fix

jeddcuison committed:
Add modify external property test

jeddcuison committed:
Refactor diagnose JVM

- Remove dead code
- Define service method as void
- Rewrite REST service accordingly due to changes in the service layer

jeddcuison commented:
This was done under a fork and since I deleted the fork this pull request won't be able to merge to master. I'm creating another pull request.
